### PR TITLE
Added date of 300th form submission

### PR DESCRIPTION
**Pull request by kkrampa**
@czue http://manage.dimagi.com/default.asp?228253#1155028

### CODE REVIEW

File: corehq/pillows/mappings/domain_mapping.py [hunk 1]
```diff
@@ -98,6 +98,8 @@
                 'cp_sms_30_d': {'type': 'boolean'},
                 'cp_last_updated': {'format': "yyyy-MM-dd||yyyy-MM-dd'T'HH:mm:ssZZ||yyyy-MM-dd'T'HH:mm:ss.SSSSSS||yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'||yyyy-MM-dd'T'HH:mm:ss'Z'||yyyy-MM-dd'T'HH:mm:ssZ||yyyy-MM-dd'T'HH:mm:ssZZ'Z'||yyyy-MM-dd'T'HH:mm:ss.SSSZZ||yyyy-MM-dd'T'HH:mm:ss||yyyy-MM-dd' 'HH:mm:ss||yyyy-MM-dd' 'HH:mm:ss.SSSSSS||mm/dd/yy' 'HH:mm:ss",
                                     'type': 'date'},
+                'cp_300th_form_submission': {'format': "yyyy-MM-dd||yyyy-MM-dd'T'HH:mm:ssZZ||yyyy-MM-dd'T'HH:mm:ss.SSSSSS||yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'||yyyy-MM-dd'T'HH:mm:ss'Z'||yyyy-MM-dd'T'HH:mm:ssZ||yyyy-MM-dd'T'HH:mm:ssZZ'Z'||yyyy-MM-dd'T'HH:mm:ss.SSSZZ||yyyy-MM-dd'T'HH:mm:ss||yyyy-MM-dd' 'HH:mm:ss||yyyy-MM-dd' 'HH:mm:ss.SSSSSS||mm/dd/yy' 'HH:mm:ss",
```
Comment (dimagimon) [inline corehq/pillows/mappings/domain_mapping.py]: E501 line too long (360 > 115 characters)

**Comment by emord** [comment]
I think you also need to change this: https://github.com/lwyszomi/commcare-hq/blob/4419c25ddd53588707852132043a6b849c1c7c50/corehq/apps/hqadmin/reports.py#L753

File: corehq/ex-submodules/couchforms/analytics.py [hunk 2]
```diff
@@ -97,6 +97,14 @@ def get_last_form_submission_received(domain):
     return submission_time
 
 
+def get_300th_form_submission_received(domain):
```
Comment (emord) [inline corehq/ex-submodules/couchforms/analytics.py]: Seems like a random place to put this method. Why not https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/domain/calculations.py? I think this is where most of the calculations like this live

**Comment by emord** [comment]
I think the totals row will be messed up without changing https://github.com/lwyszomi/commcare-hq/blob/4419c25ddd53588707852132043a6b849c1c7c50/corehq/apps/hqadmin/reports.py#L745-L762

**Pull request merged**